ebpfapi/rpc_client.cpp

```diff
@@ -1,45 +1,58 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+#include <ctype.h>
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <windows.h>
 #include "ebpf_api.h"
 #pragma warning(push)
 #pragma warning(disable : 4100) // 'identifier' : unreferenced formal parameter
 #pragma warning(disable : 4244) // 'conversion' conversion from 'type1' to
                                 // 'type2', possible loss of data
 #undef VOID
-#include "ebpf_verifier.hpp"
 #pragma warning(pop)
 #include "ebpf_windows.h"
-#include "header.h"
 #include "rpc_interface_c.c"
 
 #pragma comment(lib, "Rpcrt4.lib")
 
 static RPC_WSTR _string_binding = nullptr;
-static const WCHAR* _protocol_sequence = L"ncacn_np";
+static const WCHAR* _protocol_sequence = L"ncalrpc";
 
-#define RPC_SERVER_ENDPOINT L"\\pipe\\ebpf_service"
+#define RPC_SERVER_ENDPOINT L"ebpfsvc rpc server"
 
 int
 ebpf_rpc_verify_program(ebpf_program_verify_info* info, const char** logs, uint32_t* logs_size)
 {
-    unsigned long code;
     int result;
 
     RpcTryExcept { result = (int)ebpf_client_verify_program(info, logs_size, const_cast<char**>(logs)); }
     RpcExcept(RpcExceptionFilter(RpcExceptionCode()))
     {
-        code = RpcExceptionCode();
-        printf("ebpf_rpc_verify_program: runtime reported exception 0x%lx = %ld\n", code, code);
-        result = (int)EBPF_FAILED;
+        result = RpcExceptionCode();
     }
     RpcEndExcept
 
-        printf("ebpf_rpc_verify_program: got return code %d from the server\n\n", result);
-
     return result;
 }
 
+int
+ebpf_rpc_load_program(ebpf_program_load_info* info, const char** logs, uint32_t* logs_size)
+{
+    int result;
+
+    RpcTryExcept { result = (int)ebpf_client_verify_and_load_program(info, logs_size, const_cast<char**>(logs)); }
```

Comment:
You need to use mutual authentication to make sure you are talking to the "real" EBPF service.

Reply:
Filed #249 to track this.